src/BenchmarksServer/Startup.cs

```diff
@@ -214,13 +226,22 @@ private static string CloneRestoreAndBuild(string path, ServerJob job)
 
             AddSourceDependencies(path, benchmarksDir, dirs);
 
-            var benchmarksPath = Path.Combine(path, benchmarksDir, "src", "Benchmarks");
+            // Install latest SDK and runtime
+            // * Use custom install dir to avoid changing the default install,  which is impossible if other processes
+            //   are already using it.
+            var benchmarksRoot = Path.Combine(path, benchmarksDir);
+            ProcessUtil.Run("cmd", "/c build.cmd /t:noop", workingDirectory: benchmarksRoot,
+                environmentVariables: new Dictionary<string, string> { { "DOTNET_INSTALL_DIR", dotnetInstallDir } });
+
+            // Build and Restore
+            var benchmarksApp = Path.Combine(benchmarksRoot, "src", "Benchmarks");
+            var dotnetExecutable = Path.Combine(dotnetInstallDir, "dotnet");
 
             // Project versions must be higher than package versions to resolve those dependencies to project ones as expected.
             // Passing VersionSuffix to restore will have it append that to the version of restored projects, making them
             // higher than packages references by the same name.
-            ProcessUtil.Run("dotnet", "restore /p:VersionSuffix=zzzzz-99999", workingDirectory: benchmarksPath);
-            ProcessUtil.Run("dotnet", $"build -c Release -f {GetTFM(job.Framework)}", workingDirectory: benchmarksPath);
+            ProcessUtil.Run(dotnetExecutable, "restore /p:VersionSuffix=zzzzz-99999", workingDirectory: benchmarksApp);
+            ProcessUtil.Run(dotnetExecutable, $"build -c Release -f {GetTFM(job.Framework)}", workingDirectory: benchmarksApp);
```

Comment:
Is the intention here to use the latest-and-greatest `dotnet` when running the benchmark? Or, should the solution configuration in the `source` commit be able to choose a specific version?

If the second, better to execute a single `build.cmd` / `build.sh` command which installs the necessary `dotnet`, restores one project, builds that project, then runs that project. Need a new target or three in KoreBuild for this.